src/molecule.h

```diff
@@ -134,24 +137,30 @@ class Molecule : protected Pointers {
  private:
   int me;
   FILE *fp;
+  json jsonf;
   int *count;
   int toffset, boffset, aoffset, doffset, ioffset;
   int autospecial;
   double sizescale;
 
   void read(int);
+  void read_json(int);
```

Comment:
In a hypothetical MoleculeJSON class, you could, derive it from Molecule. Flag the `read()`, `coords()` and similar methods as virtual in the base class and then as override in the derived class. I am currently not looking at the implementation in detail, just for general formal issues with contributions as documented in the manual and established as common practices, so there may be potential for more code sharing by separating parsing and and application of parsed data. But that is for a future revision.